### DISCUSSION TITLE
Generate report only if build was successfull

### DISCUSSION THREAD

**Discussion by alphanull** [webpack-contrib/webpack-bundle-analyzer#585] [Ideas]
Right now, when my build fails the bundle analyzer runs nonetheless, spitting out hundreds of lines like "Error parsing bundle asset ..."

It would be much nicer if I could configure this plugin to run only after a **successful** build

**Comment by valscion** [comment]
Hmm but what if only some part was unsuccessful but the output still contains valid bundles? Is there a hybrid scenario possible where only some part of the webpack build failed but `webpack-bundle-analyzer` could still work correctly?

**Comment by alphanull** [comment]
At least in my case, when my build fails no files are generated at all. And since I clear my dist folder before each build, I get those messages I mentioned:

    Error parsing bundle asset "/path/to/file.js": no such file
And at the end:

    No bundles were parsed. Analyzer will show only original module sizes from stats file.
Maybe we have another problem, I mean why is this plugin trying to analyze those non existing files in the first place?

**Comment by valscion** [comment]
It is because the [webpack stats object](https://webpack.js.org/api/stats/) generated by webpack lists the bundles that would be output and `webpack-bundle-analyzer` tries to then parse those files.

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/BundleAnalyzerPlugin.js#L33-L38

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/BundleAnalyzerPlugin.js#L54

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/BundleAnalyzerPlugin.js#L119-L120

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/viewer.js#L117

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/viewer.js#L127

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/viewer.js#L189-L199

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/analyzer.js#L20

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/analyzer.js#L50-L60

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/analyzer.js#L9

https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/34b6223bcb25292f85fb271d0a1208b2f4a2d0ec/src/parseUtils.js#L10-L11

**Comment by th0r** [comment]
We can probably do nothing if there are compilation errors? @valscion what do you think?

**Comment by valscion** [comment]
> We can probably do nothing if there are compilation errors? @valscion what do you think?

Yeah, that might make sense. It's likely that we won't be able to do much with the resulting bundles (if there are any) after compilation errors anyway.

**Comment by alphanull** [comment]
Yeah, doing nothing after any errors (including errors resulting from multiconfigs) would be nice. Of course this also could be a configuration setting, so if users want to have a report even after having errors, they still can do that. But I think in most cases, it makes no sense to generate a report when the build fails anyways.

**Comment by valscion** [comment]
Nah, I'd prefer not adding new configuration options for this

**Comment by alphanull** [comment]
OK, I also would be fine with no additional options

**Comment by thomas-darling** [comment]
This still appears to be an issue.
It really makes no sense to generate a report for a failed build, and those messages create a lot of noise in the logs, which distracts from the real errors. Any chance we could have those messages suppressed when the build fails?

**Comment by fregante** [comment]
Same here, it's incredibly annoying to see `webpack-bundle-analyzer` pile on webpack errors _and_ still open a useless report file in the browser.

By default it should just detect the failure and output just **"Webpack failed, no soup for you!"**, especially when it cannot find the assets at all and it shows:

```
Error parsing bundle asset "/path/to/file.js": no such file
```

**Comment by snitin315** [comment]
PR welcome :)
